### PR TITLE
front: order electrical profils

### LINKED ISSUE #5412: Electrical profile O is not put in the right order
When we enter the profile O, this is put in the end of the list.

Image

It should be before the A profile.

version 358b34a7

### LINKED ISSUE #5562: Electrical profiles are not always in the right order
### What happened?

The electrical profiles for a 25kV rolling stock appear in this order :
image

### What did you expect to happen?

They should be following the same logic as the 1500V ones, which is the strongest above. So this means
- 25000
- 22500
- 20000

And for 1500V, the order must be:
- O
- A
- A1
- B
- B1
- C
- D
- E
- F
- G

### How can we reproduce it (as minimally and precisely as possible)?

1. Go to RS editor
2. Duplicate a RS with 25kV voltage
3. Go to the speed-effort curves editor

### What operating system, browser and environment are you using?

- **Browser**: Chrome
- **OS**: W11
- **Env**: dev

### OSRD version (top right corner `Account` button > `Informations`)

8ecea1c4

### PR DESCRIPTION
**Pull request by SarahBellaha**
Close #5412
Close #5562 (same issue)

### CODE REVIEW

File: front/src/modules/rollingStock/consts.ts [hunk 1]
```diff
@@ -336,8 +336,28 @@ export type ElectricalProfileByMode = {
...
+    'H',
+    'I',
+    'J',
+    'K',
+    'L',
+    'M',
+    'N',
+    'O',
+  ],
+  '25000': [null, '25000', '22500', '20000'],
```
Comment (Yohh) [inline front/src/modules/rollingStock/consts.ts:396]: are you sure those profiles really exist?

Comment (SarahBellaha) [reply front/src/modules/rollingStock/consts.ts:396]: Yes 👍

**Review by Yohh — APPROVED** [review]
LGTM :muscle:

File: front/src/modules/rollingStock/helpers/utils.ts [hunk 2]
```diff
@@ -261,3 +262,16 @@ export const sortSelectedModeCurves = (curvesList: ConditionalEffortCurve[]) =>
...
+  electricalProfilesList: (string | null)[],
+  selectedTractionMode: string | null
+) => {
+  const isSelectedModeWithProfiles =
+    selectedTractionMode === '1500' || selectedTractionMode === '25000';
+  const refArray = isSelectedModeWithProfiles
+    ? electricalProfilesByMode[selectedTractionMode]
+    : [null];
+
+  return electricalProfilesList.sort((a, b) => refArray.indexOf(a) - refArray.indexOf(b));
```
Comment (anisometropie) [inline front/src/modules/rollingStock/helpers/utils.ts]:
```suggestion
  if (isSelectedModeWithProfiles) {
    const refArray = electricalProfilesByMode[selectedTractionMode];
    return electricalProfilesList.sort((a, b) => refArray.indexOf(a) - refArray.indexOf(b));
  }
  return electricalProfilesList;
```

**Review by anisometropie — CHANGES_REQUESTED** [review]

File: front/src/modules/rollingStock/helpers/utils.ts [hunk 3]
```diff
@@ -261,3 +262,16 @@ export const sortSelectedModeCurves = (curvesList: ConditionalEffortCurve[]) =>
 
 export const isElectric = (rollingStock: RollingStock) =>
   some(Object.values(rollingStock.effort_curves.modes), (effortCurve) => effortCurve.is_electric);
+
+export const orderElectricalProfils = (
```
Comment (anisometropie) [inline front/src/modules/rollingStock/helpers/utils.ts:265]: can you write a very basic test

File: front/src/modules/rollingStock/consts.ts [hunk 4]
```diff
@@ -336,8 +336,28 @@ export type ElectricalProfileByMode = {
...
+    'F',
+    'G',
+    'H',
+    'I',
+    'J',
+    'K',
+    'L',
+    'M',
+    'N',
+    'O',
```
Comment (axrolld) [inline front/src/modules/rollingStock/consts.ts]:
```suggestion
    
```
`O` is the best profile, it shouldn't be at the end of the list

**Review by axrolld — CHANGES_REQUESTED** [review]

**Review by axrolld — APPROVED** [review]
Lgtm 🎉

**Review by anisometropie — APPROVED** [review]

File: front/src/modules/rollingStock/components/RollingStockEditor/__tests__/RollingStockEditorCurves.spec.ts
Comment (clarani) [inline front/src/modules/rollingStock/components/RollingStockEditor/__tests__/RollingStockEditorCurves.spec.ts:1]: Super nice the test, thanks !!

**Review by clarani — APPROVED** [review]
LGTM, thanks :white_check_mark: Tested !

**Pull request merged**